lib/recorder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <memory>
+
+namespace swss {
+
+class RecBase {
+public:
+    RecBase() {}
+    /* Setters */
+    void setRecord(bool record)  { enable_rec = record; }
+    void setRotate(bool rotate)  { log_rotate = rotate; }
+    void setLocation(const std::string& loc) { location = loc; }
+    void setFileName(const std::string& name) { filename = name; }
+    void setName(const std::string& name)  { m_name = name; }
+
+    /* getters */
+    bool isRecord()  { return enable_rec; }
+    bool isRotate()  { return log_rotate; }
+    std::string getLoc() { return location; }
+    std::string getFile() { return filename; }
+    std::string getName() { return m_name; }
+
+private:
+    bool enable_rec;
```

Comment:
>enable_rec [](http://example.com/codeflow?start=9&length=10)

Please add verbose documents to new code. For example, what is `m_name`, if it is a name, why you have another `filename`. Seems like `m_name` should be `m_somename`.

Reply:
m_name is just the user readable name of the type of recorder. "SwSS", "Response Publisher" & "SaiRedis". Used  while printing the logs

Eg: 

`SWSS_LOG_NOTICE("%s Recorder: Recording started at %s", getName().c_str(), fname.c_str());`

`Jun 23 19:26:15.741815 sonic NOTICE swss#orchagent: :- startRec: SwSS Recorder: Recording started at /var/log/swss/swss.rec`